app/src/main/java/com/mapbox/maps/testapp/examples/viewport/AdvancedViewportGesturesExample.kt

```diff
@@ -0,0 +1,194 @@
+        AnnotationUtils.loadStringFromAssets(
+          this,
+          NAVIGATION_ROUTE_JSON_NAME
+        )
+      ).routes()[0].geometry()!!,
+      Constants.PRECISION_6
+    )
+    mapView.getMapboxMap().loadStyle(
+      // Show the route line on the map
+      style(Style.TRAFFIC_DAY) {
+        +geoJsonSource(GEOJSON_SOURCE_ID) {
+          geometry(routePoints)
+        }
+        +lineLayer(
+          ROUTE_LINE_LAYER_ID,
+          GEOJSON_SOURCE_ID
+        ) {
+          lineColor(mapView.context.resources.getColor(R.color.mapbox_blue))
+          lineWidth(10.0)
+          lineCap(LineCap.ROUND)
+          lineJoin(LineJoin.ROUND)
+        }
+      }
+    ) {
+      setupLocationComponent(routePoints)
+      setupAdvancedGestures()
+      mapView.experimentalViewport.transitionTo(overviewViewportState)
+    }
+  }
+
+  private fun setupAdvancedGestures() {
+    // Disable viewport's transitionsToIdleUponUserInteraction so that we can customise the gestures
+    // within the state.
+    mapView.experimentalViewport.transitionsToIdleUponUserInteraction = false
+
+    // Advanced gestures handling
+
+    // Disable scroll gesture
+    mapView.gestures.scrollEnabled = false
+
+    // Switch ViewportStates by single tapping on the map.
+    mapView.gestures.addOnMapClickListener {
+      mapView.experimentalViewport.transitionTo(
+        when (mapView.experimentalViewport.status.getCurrentOrNextState()) {
+          followingViewportState -> overviewViewportState
+          else -> followingViewportState
+        }
+      )
+      false
+    }
+
+    // Let shove gesture to overwrite the followingViewportState's default pitch value.
+    mapView.gestures.addOnShoveListener(
+      object : OnShoveListener {
+        override fun onShoveBegin(detector: ShoveGestureDetector) {
+          // no-ops
+        }
+
+        override fun onShove(detector: ShoveGestureDetector) {
+          followingViewportState.options = followingViewportState.options.toBuilder()
+            .defaultPitch(followingViewportState.options.defaultPitch - PITCH_FACTOR * detector.deltaPixelSinceLast)
+            .build()
+        }
+
+        override fun onShoveEnd(detector: ShoveGestureDetector) {
+          // no-ops
+        }
+      })
+
+    // Let scale gesture to overwrite the followingViewportState's default zoom value.
+    mapView.gestures.addOnScaleListener(object : OnScaleListener {
+      override fun onScaleBegin(detector: StandardScaleGestureDetector) {
+        // no-ops
+      }
+
+      override fun onScale(detector: StandardScaleGestureDetector) {
+        val zoomBy = ln(detector.scaleFactor.toDouble()) / ln(PI / 2) * SCALE_FACTOR
+        followingViewportState.options = followingViewportState.options.toBuilder()
+          .defaultZoom(followingViewportState.options.defaultZoom + zoomBy).build()
+      }
+
+      override fun onScaleEnd(detector: StandardScaleGestureDetector) {
+        // no-ops
+      }
+    })
+  }
```

Comment:
@macdrevx @LukasPaczos I added a simple example to demonstrate how to customise the gestures to work with the FollowingViewportStates.